keanu-project/src/main/java/io/improbable/keanu/tensor/intgr/Nd4jIntegerTensor.java

```diff
@@ -368,7 +377,8 @@ public IntegerTensor divInPlace(IntegerTensor that) {
         if (that.isScalar()) {
             tensor.divi(that.scalar());
         } else {
-            INDArrayShim.divi(tensor, unsafeGetNd4J(that), tensor);
+            INDArray result = INDArrayShim.divi(tensor, unsafeGetNd4J(that));
+            return new Nd4jIntegerTensor(INDArrayExtensions.castToInteger(result, false));
```

Comment:
Do we need the "if result != tensor" again here?

Reply:
good catch!!!